pkg/expansion/system_test.go

```diff
@@ -83,10 +83,21 @@ func TestExpand(t *testing.T) {
 		},
 		{
 			name:      "expand with nil namespace returns error",
-			generator: fixtures.LoadFixture(fixtures.DeploymentNginx, t),
+			generator: fixtures.LoadFixture(fixtures.DeploymentNginxWithNs, t),
 			ns:        nil,
 			mutators: []types.Mutator{
-				fixtures.LoadAssign(fixtures.AssignPullImage, t),
+				fixtures.LoadAssign(fixtures.AssignPullImageWithNs, t),
+			},
+			templates: []*expansionunversioned.ExpansionTemplate{
+				fixtures.LoadTemplate(fixtures.TempExpDeploymentExpandsPods, t),
+			},
+			expectErr: true,
```

Comment:
I maybe not be reading this right, but why does this result in error when the deployment/parent has the namespace?

Reply:
great question! on line `87` see that the namespace is still `nil`. That is the `namespace` object we'd pass in to the expansion system.

so before this PR, if the namespace on line 87 was nil, it would always fail. Now it only fails if there is a namespace selector, like the mutator in `fixtures.AssignPullImageWithNs`. 

then the `matcher` code will err out, so that's why we still expect an error